astropy/modeling/core.py

```diff
@@ -2315,6 +2410,159 @@ def _custom_model_wrapper(func, fit_deriv=None):
+                        x_stddev = np.abs(np.random.rand() + 5),
+                        y_stddev = np.abs(np.random.rand() * 8 + 5),
+                        theta = np.random.rand() * np.pi) for i in range(nsrc)]
+
+        model_list = [Gaussian2D(**kwargs) for kwargs in model_params]
+
+        bb_array = np.zeros(imshape)
+        full_array = np.zeros(imshape)
+
+        for model in model_list:
+            render_model(model, bb_array, copy=False)
+            model.bounding_box = None
+            render_model(model, full_array, copy=False)
+
+        max_err = (full_array-bb_array).max()
+
+        plt.figure(figsize=(12,8))
+        plt.imshow(np.sqrt(bb_array), cmap='cubehelix')
+        plt.colorbar()
+        plt.title('Bounded Models', fontsize=16)
+
+        plt.figure(figsize=(12,8))
+        plt.imshow(full_array - bb_array, cmap='RdBu', vmin=-max_err, vmax=max_err)
+        plt.colorbar(format='%.1e')
+        plt.title('Total Flux Err = %.0e' % 
+                  ((np.sum(full_array) - np.sum(bb_array))/np.sum(full_array)),
+                  fontsize=16)
+        plt.show()
+    """
+
+    bbox = model.bounding_box
+
+    if (coords is None) & (arr is None) & (bbox is None):
+        raise AssertionError('If no bounding_box is set, coords or arr must be input.')
+
+    # for consistent indexing
+    if model.n_inputs == 1:
+        if coords is not None:
+            coords = [coords]
+        if bbox is not None:
+            bbox = [bbox]
+
+    if arr is not None:
+        # Check dimensions match model
+        assert arr.ndim == model.n_inputs
+        # Copy
+        if copy:
+            arr = arr.copy()
+
+    if coords is not None:
+        # Will fail if coords are inconsisent
+        coords = np.array(coords)
+
+        # Check dimensions match model
+        assert len(coords) == model.n_inputs
+
+        # Check dimensions match arr
+        if arr is not None:
+            assert coords[0].shape == arr.shape
+
+        # If no input arr, make one
+        else:
+            arr = np.zeros(coords[0].shape)
+
+    if bbox is not None:
+
+        # assure position is at center pixel
+        pd = pos, delta = np.array([(np.mean(bb), np.ceil((bb[1] - bb[0]) / 2))
+                                    for bb in bbox]).astype(int).T
+
+        if coords is not None:
+
+            sub_shape = tuple(delta * 2 + 1)
+            sub_coords = np.array([extract_array(c, sub_shape, pos) for c in coords])
+
+        else:
+
+            limits = [slice(p - d, p + d + 1, 1) for p, d in pd.T]
+
+            sub_coords = np.mgrid[limits]
+
+        if model.n_inputs > 1:
+            sub_coords[[0, 1]] = sub_coords[[1, 0]]
+
+        if arr is None:
+            arr = model(*sub_coords)
+
+        else:
+            try:
+                arr = add_array(arr, model(*sub_coords), pos)
+            except ValueError:
+                raise ValueError('The `bounding_box` is larger than the input arr in one \
```

Comment:
Does this work, adding a `\` within a multi-line string?
In any case, I think the common pattern is
```python
raise ValueError('my very '
                 'long string')
```